realsense2_camera/src/ros_sensor.cpp

```diff
@@ -18,6 +18,22 @@
 using namespace realsense2_camera;
 using namespace rs2;
 
+#define SAFE_SET_OPTION(opt, val)                                    \
+    if (this->supports(opt))                                          \
+    {                                                                \
+        try                                                          \
+        {                                                            \
+            this->set_option(opt, val);                               \
+        }                                                            \
+        catch(std::exception& e)                                     \
+        {                                                            \
+            ROS_WARN_STREAM("Failed to set option: " << e.what());   \
```

Comment:
Can you please add more info?
Something like
```
ROS_WARN_STREAM("Failed to set option: " << opt << " to val: " << val << " error: " e.what());
```

(just make sure if some casting is needed)